### ISSUE TITLE
Alerts - color contrast of icons

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly-next#1474]
The icons used for Alerts should be double-checked for color contrast. A quick check shows that the danger color combination is not accessible. These icons should have contrast given that for users who are color blind, this is the means by which they would understand the type of alert.

**Comment by jgiardino** [comment]
@kybaker I'm not sure if this issue would be addressed with the color updates you're working on or not, but wanted to go ahead and capture it.

**Comment by mcoker** [comment]
@jgiardino do you mind checking this again since we've made the updates? https://pf4.patternfly.org/components/Alert/examples/

**Comment by jgiardino** [comment]
These are the results of the color contrast for icons in Alerts:

Info icon on background - [pass](https://contrast-ratio.com/#rgb%280%2C%2067%2C%20104%29-on-rgb%28115%2C%20188%2C%20247%29)
Success icon on background - [pass](https://contrast-ratio.com/#rgb%2872%2C%20107%2C%200%29-on-rgb%28146%2C%20212%2C%200%29)
Warning icon on background - [pass](https://contrast-ratio.com/#rgb%28121%2C%2086%2C%200%29%20-on-rgb%28240%2C%20171%2C%200%29)
Danger icon on background - [fail](https://contrast-ratio.com/#rgb%2871%2C%200%2C%200%29-on-rgb%28201%2C%2025%2C%2011%29) 👎

**Comment by mcoker** [comment]
Thanks! @kybaker or @mceledonia should we make an update here?

**Comment by mceledonia** [comment]
@kybaker I heard the alert design might be changing to align with the digital design team design, do we have any updates there? 

If not then yes we should change the danger icon on background to achieve a pass result for contrast.

**Comment by kybaker** [comment]
@mceledonia We should just change the icon color for now.

**Comment by mceledonia** [comment]
@jgiardino Are we able to subvert the a11y requirement by noting that the icon itself is more decorative and doesn't convey new information that the alert wouldn't have without it?

**Comment by jgiardino** [comment]
@mceledonia I was seeing the icon as the alternate means of communication for color blind users who can't perceive the colors we use to indicate alert type. We do have a screen reader alternative for the icon and color, but for users who are color blind, I would expect there to be some accessible identifier for the type of alert without needing to read the contents to discern if the alert is an error or some other type. I was thinking the icon would serve this purpose, given that there's sufficient color contrast.

**Comment by mceledonia** [comment]
Here is a solution for this one @jgiardino @kybaker @mcoker 

We switch our danger-100 and -200 (which really means switching our red-100 and red-200) to the appropriate brand red colors.

Then we switch danger-300 to use red-500, and change red-500 to #120100 which gives us a perfect 4.5 on top of our new danger-100.

![image](https://user-images.githubusercontent.com/30935938/66512451-f658db80-eaa6-11e9-91d2-aba473769530.png)

**Comment by jgiardino** [comment]
I was chatting with @mceledonia offline about this and want to capture it here for everyone. I'm good with whatever decisions you make for colors, assuming they meet color contrast.

I'm inclined to say that for Alerts (not inline alerts), we can use the 3:1 color ratio, based on this note included in the description for the [WCAG 1.4.3 color contrast criteria](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html):
> When evaluating this success criterion, the font size in points should be obtained from the user agent or calculated on font metrics in the way that user agents do. Point sizes are based on the CSS pt size as defined in CSS3 Values. The ratio between sizes in points and CSS pixels is 1pt = 1.333px, therefore 14pt and 18pt are equivalent to approximately **18.5px** and **24px**.

I'm assuming that the two pixel sizes stated above can be applied to our icons. This means that 18.5px plus bold or 24px not bold can use 3:1. It's debatable whether the icons qualify as "bold" or not, and I'd defer to the visual designers on that. But if we were to be conservative and say that the icons would not qualify as bold, then this would mean that the icons in Alerts that are 24px can be 3:1, but other icons that are smaller, like in the Inline Alerts, would still need to be 4.5:1.

I _think_ all of the smaller icons are always on white? So hopefully these sizes and ratios can allow the visual designers to get the color palette they want?

**Comment by mceledonia** [comment]
Going to close this one, I'm planning on redesigning the toast alerts to not have the color icon over color background, making any accessibility requirements much easier to achieve.
